gooddata-sdk/gooddata_sdk/catalog/declarative/dataset.py

```diff
@@ -0,0 +1,307 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any
+
+from gooddata_metadata_client.model.data_source_table_identifier import DataSourceTableIdentifier
+from gooddata_metadata_client.model.declarative_attribute import DeclarativeAttribute
+from gooddata_metadata_client.model.declarative_dataset import DeclarativeDataset
+from gooddata_metadata_client.model.declarative_date_dataset import DeclarativeDateDataset
+from gooddata_metadata_client.model.declarative_fact import DeclarativeFact
+from gooddata_metadata_client.model.declarative_label import DeclarativeLabel
+from gooddata_metadata_client.model.granularities_formatting import GranularitiesFormatting
+from gooddata_sdk.catalog.declarative.identifier import CatalogGrainIdentifier
+from gooddata_sdk.catalog.declarative.reference import CatalogDeclarativeReference
+from gooddata_sdk.catalog.entity import CatalogTitleEntity
+
+
+class CatalogDeclarativeDataset(CatalogTitleEntity):
+    def __init__(
+        self,
+        id: str,
+        title: str,
+        grain: list[CatalogGrainIdentifier],
+        references: list[CatalogDeclarativeReference],
+        description: str = None,
+        attributes: list[CatalogDeclarativeAttribute] = None,
```

Comment:
again:
default [], please, we anyway have to inject e.g. attributes key into the payload